modules/container_definition/environment.tf

```diff
@@ -0,0 +1,14 @@
+locals {
+  environment_keys        = keys(var.environment)
+  sorted_environment_keys = sort(local.environment_keys)
+
+  sorted_environment = [
+    for key in local.sorted_environment_keys :
+    {
+      name  = key
+      value = lookup(var.environment, key)
+    }
+  ]
+
+  final_environment = length(local.sorted_environment_keys) > 0 ? local.sorted_environment : []
```

Comment:
Can this not be just `local.sorted_environment` all the time or does TF not properly map over arrays?

Reply:
Yes, it can. Good catch.